spec/requests/users_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+RSpec.describe "/users", type: :request do
+  
+  # This should return the minimal set of attributes required to create a valid
+  # User. As you add validations to User, be sure to
+  # adjust the attributes here as well.
+  let(:valid_attributes) {
+    skip("Add a hash of attributes valid for your model")
```

Comment:
It doesn't look like any of the tests have been filled out in this file, other than `GET /new`. They are quite important to keep bugs out of the controller.